### ISSUE TITLE
 thread 'main' has overflowed its stack

### ISSUE THREAD

**Issue by steveklabnik** [kbknapp/cargo-outdated#68]
This happens whenever I run `cargo-outdated` in the Rust repository's src directory.

Rust 1.20.

**Comment by Frederick888** [comment]
@steveklabnik I tested it myself and got

```
⇒  cargo outdated -v
  Parsing... current workspace
Resolving... current workspace
error: manifest path `/home/frederick/Programming/Rust/rust/src/Cargo.toml` is a virtual manifest, but this command requires running against an actual package in this workspace
```

...since #58 looping through workspace members is not supported yet. Which version of `cargo-outdated` were you using?

**Comment by steveklabnik** [comment]
```bash
> cargo outdated --version
cargo-outdated v0.4.0
```

I had just run 'cargo-install'.

**Comment by Frederick888** [comment]
@steveklabnik Well, I don't have the permission to publish the new `0.5.0` to crates.io, so you could only manually clone this repo and run `cargo install --path . -f`.

But even though since I haven't got time to do #58 , it cannot be used on a virtual manifest out of the box. Maybe a simple Bash script could do the job for now.

**Comment by steveklabnik** [comment]
>  I don't have the permission to publish the new 0.5.0 to crates.io, 

Ah interesting, thanks! (Also, `cargo install` supports a `--git` flag, so that'd be even easier to install).

This seems bad though; what happened? I really love cargo-outdated.

How hard do you think #58 is?

**Comment by Frederick888** [comment]
@steveklabnik @kbknapp has been a bit busy recently, let's just wait him to be back patiently :)

I don't think #58 would be too hard. Basically it would be a matter of replacing `workspace.current()` calls with some workspace-member-aware logics and finally looping through all the members to compare dependency trees and print the results out... some extra work will be needed to reused the temporary workspace though.

**Comment by kbknapp** [comment]
My apologies! Yeah my time has been monopolized lately, @Frederick888 I'll add you so you can publish new version as well.

**Comment by kbknapp** [comment]
@Frederick888 you should be able to publish new versions to crates.io now as well. Aside, feel free to add yourself to the authors array as well, I think you've earned it :wink:

**Comment by steveklabnik** [comment]
> @steveklabnik @kbknapp has been a bit busy recently, let's just wait him to be back patiently :)

Ah, it's all good! Everyone needs some time off. Looks like @kbknapp came back briefly between now and then; thank you for one of the most useful cargo commands!

**Comment by Frederick888** [comment]
@kbknapp Many thanks! I have just published `v0.5.1` with some minor changes since the last release.

I think I'm going to close this issue in support of #58. And by the way, I have assigned #58 a P1 label and I'd work on it first if I can make some time. My uni assessments are killing me right now lol...
